File: week-6---pointers/Lecture/lec_some_function.cpp
#include <iostream>
#include <array>
using namespace std;

void some_function( array<int,5> values )
{
	for( int i=0; i<values.size(); ++i )
		values[i] *= 2;
}		

int main()
{
	array<int,5> v {0, 1, 2, 3, 4};

	for( int i : v )		// 0,1,2,3,4
		cout << i << ",";
	cout << endl;

	some_function(v);
	for( int i : v )		// 0,1,2,3,4
		cout << i << ",";
	cout << endl;
}

File: week-6---pointers/Lecture/lec_some_function.py
import sys

def some_function( values ):
	for i in range(len(values)):
		values[i] *= 2

def main():
	v = [ i for i in range(5) ]
	print(v)  # [0, 1, 2, 3, 4]

	some_function(v)
	print(v)  # [0, 2, 4, 6, 8]

if __name__ == '__main__':
	sys.exit(main())